src/Microsoft.AspNet.OData.Shared/Builder/PropertyConfiguration.cs

```diff
@@ -157,6 +164,17 @@ public bool Unsortable
         /// </summary>
         public QueryConfiguration QueryConfiguration { get; set; }
 
+        /// <summary>
+        /// Adds subtypes to the set of derived type constraints.
+        /// </summary>
+        internal void AddDerivedTypeConstraintImpl(params Type[] subtypes)
+        {
+            foreach (Type t in subtypes)
+            {
+                DerivedTypeConstraints.ValidateAndAddConstraint(RelatedClrType, t);
```

Comment:
>ValidateAndAddConstraint [](start = 39, length = 24)

Maybe define a ValidateAndAddConstraints(Type RelatedClrType, params Type[] subtypes) method to DerivedTypeConstraints class?